pkg/schemadsl/compiler/type_annotations_invalid_test.go

```diff
@@ -0,0 +1,82 @@
+package compiler
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/authzed/spicedb/pkg/schemadsl/input"
+)
+
+func TestPermissionTypeAnnotationInvalidSyntax(t *testing.T) {
```

Comment:
no need for this test suite; instead, add `.zed` files with these parsing errors as test cases